code/modules/food_and_drink/cakes.dm

```diff
@@ -11,75 +27,491 @@
+		if(tube.reagents.total_volume < 25)
+			user.show_text("The icing tube isn't full enough to frost the cake!","red")
+			return
+		var/frostingtype
+		frostingtype = input("Which frosting style would you like?", "Frosting Style", null) as null|anything in frostingstyles
+		if(frostingtype && (get_dist(src, usr) <= 1))
+			var/tag
+			var/datum/color/average = tube.reagents.get_average_color()
+			switch(frostingtype)
+				if("classic")
+					tag = "cake[clayer]-classic"
+				if("top swirls")
+					tag = "cake[clayer]-swirl_top"
+				if("bottom swirls")
+					tag = "cake[clayer]-swirl_bottom"
+				if("spirals")
+					tag = "cake[clayer]-spiral"
+				if("rose spirals")
+					tag = "cake[src.clayer]-spiral_rose"
+			if(!src.GetOverlayImage(tag))
+				if(src.sliced)
+					tag = replacetext(tag,"cake[clayer]","slice")
+				var/image/frostingoverlay = new /image(src.icon,tag)
+				frostingoverlay.color = average.to_rgba()
+				frostingoverlay.alpha = 255
+				src.UpdateOverlays(frostingoverlay,tag)
+				tube.reagents.trans_to(src,25)
+
+	proc/overlay_number_convert(var/original_clayer,var/mode,var/singlecake) //original - original clayer value, mode - which math we're using
+		switch(mode)
+			if(CAKE_MODE_STACK)
+				if(original_clayer==2) //if the cake is two cakes high now
+					. = list(1,2) //replacetext all instances of 1 with 2, updating the first layer overlays with second layer overlays
+				else if(original_clayer==3 && singlecake) //if the cake is three cakes high now and the player is adding a single cake to it
+					. = list(1,3) //replace all instances of 1 with 3, updating the first layer overlays with third layer overlays
+				else //fringe case catch yay math
+					. = list(2,3)
+			if(CAKE_MODE_BUILD) //same idea, but different math
+				if(original_clayer==2)
+					. = list(2,1)
+				else if(original_clayer==3)
+					. = list(3,1)
+
+
+	proc/slice_cake(var/obj/item/W,var/mob/user)
+		if (src.sliced)
+			user.show_text("This cake has already been sliced!","red")
+			return
+		user.show_text("You cut the cake into slices.")
+		var/layer_tag //passes layer information to the build_cake function
+		var/replacetext //used to change layer identifiers to reformat from cake overlays to slice overlays
+		var/obj/item/reagent_containers/food/snacks/cake/s = new /obj/item/reagent_containers/food/snacks/cake //temporary reference item to paste overlays onto child items
+		var/candle_lit
+		switch(src.clayer) //checking the current layer of the cake
+			if(1)
+				layer_tag = "first" //the tag of the future overlay
+				replacetext = "cake1" //used in replacetext below to assign overlays
+			if(2)
+				layer_tag = "second"
+				replacetext = "cake2"
+			if(3)
+				layer_tag = "third"
+				replacetext = "cake3"
+
+		candle_lit = build_cake(s,user,CAKE_MODE_SLICE,layer_tag,replacetext)
+
+		var/transferamount = (src.reagents.total_volume/10)/clayer //amount of reagent to transfer to slices
```

Comment:
I feel like the `10` here and lower could be a define or a var on the cake. Just in case someone wants to make 30 slice cakes or something later and also for clarity.

Reply:
good call

Reply:
so...i did a little bit of thinking and this would fit muuuuch better in the followup patch i have planned that adds extra QoL to cakes and such for edge cases and coder-friendliness to build onto.... would you mind if i added this to the followup to-do list? if i added it now, i'd have to move a bit of core cake code around which would inflate this single patch even more.

Reply:
i mean, it's just a `#define CAKE_SLICES 10`......

Reply:
it's not just that...because say you have a custom cake with 2 slices and you stack it on top of another cake. the fact that the second cake in the stack only has 2 slices has to be stored somewhere and it currently isn't built to handle that, so i'm planning on adding this functionality when i do thinks like layer descriptions and names.

Reply:
i added a define for now